types/mempool/sender_nonce.go

```diff
@@ -154,6 +176,12 @@ func (snm *senderNonceMempool) Remove(tx sdk.Tx) error {
 	if senderTxs.Len() == 0 {
 		delete(snm.senders, sender)
 	}
+	key := txKey{nonce: nonce, address: sender}
+	_, found = snm.existingTx[key]
+	if found {
+		delete(snm.existingTx, key)
+	}
+
 	return nil
```

Comment:
Simply
```go
delete(snm.existingTx, key)
```

unless we wanted to ONLY delete and return nil if it exists otherwise return an error but I think the error should only be predicated upon not having at least one signer so just `delete(map, key)` suffices